evm/chains/chain.py

```diff
@@ -142,6 +146,15 @@ def create_header_from_parent(self, parent_header, **header_params):
             block_number=parent_header.block_number + 1,
         ).create_header_from_parent(parent_header, **header_params)
 
+    def set_gas_estimator(self, gas_estimator_func):
```

Comment:
I'm not sure this needs a setter.  I'd prefer this be something that is configured at the class level rather than the instance level.  In addition, I'd like it if it was configurable using environment variables (like the database backend).

Something like:

```python
class Chain:
    gas_estimation_backend = None

    def get_gas_estimation_backend(self):
        if self.gas_estimation_backend is None:
            # check whether a path has been declared using an environment variable.  If not, return the "default" backend.
            ...
        return self.gas_estimation_backend()
```

Whatcha think?